js/foundation.offcanvas.js

```diff
@@ -136,6 +137,20 @@ class OffCanvas extends Plugin {
       this.$element.css('transition-duration', this.options.transitionTime);
     }
 
+    let inCanvasFor = this.$element.attr('class').match(/\bin-canvas-for-(\w+)/);
+
+    if (!this.options.inCanvasOn && inCanvasFor && inCanvasFor.length === 2) {
+      // Extract breakpoint name from CSS class if no inCanvasOn option set but using .in-canvas-for-[BREAKPOINT]
+      this.options.inCanvasOn = inCanvasFor[1];
+    } else if (this.options.inCanvasOn && !inCanvasFor) {
+      // Add CSS class if only inCanvasOn option set
+      this.$element.addClass(`in-canvas-for-${this.options.inCanvasOn}`);
+    }
+
+    if (this.options.inCanvasOn) {
+      this.isInCanvas = MediaQuery.atLeast(this.options.inCanvasOn);
+    }
```

Comment:
Duplicated with `:178`. You can move this to its own function like `_reflowBreakpoint`.

Reply:
Right, code should be DRY. I've now moved the code into a new function called `_checkInCanvas` because this is most clear imo.